src/game/91F8C.c

```diff
@@ -50,7 +66,38 @@ INCLUDE_ASM("asm/game/nonmatchings/91F8C", func_800B5858);
 
 INCLUDE_ASM("asm/game/nonmatchings/91F8C", func_800B5868);
 
+// problem with nop-s after this function
+#ifndef NON_MATCHING
 INCLUDE_ASM("asm/game/nonmatchings/91F8C", func_800B5878);
+#else
+u8* func_800B5878(u8* arg0, u8* arg1, s32 arg2) {
+    s32 var_a2;
+    u8 temp_v0;
+    u8* temp_v1;
+    u8* var_a0;
+    u8* var_a1;
+    u8* var_v0;
+
+    var_a0 = arg0;
+    var_a1 = arg1;
+    var_a2 = arg2;
+    var_v0 = NULL;
+    if (var_a0 != NULL) {
+        temp_v1 = var_a0;
+        if (var_a2 > 0) {
+            do {
+                temp_v0 = *var_a1;
+                var_a1 += 1;
+                var_a2 -= 1;
+                *var_a0 = temp_v0;
+                var_a0 += 1;
+            } while (var_a2 > 0);
+        }
+        var_v0 = temp_v1;
+    }
+    return var_v0;
+}
```

Comment:
```suggestion
u8* func_800B5878(u8* arg0, u8* arg1, s32 arg2) {
    u8* ret = NULL;
    u8* temp_v1;
    
    if (arg0 != NULL) {
        temp_v1 = arg0;
        while (arg2 > 0) {
            *arg0 = *arg1;
            arg1++;
            arg2--;
            arg0++;
        }
        ret = temp_v1;
    }
    return ret;
}
```
simplified this function a bit